src/node/entity_id.h

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+#pragma once
+
+#include "ds/json.h"
+
+#include <fmt/format.h>
+#include <msgpack/msgpack.hpp>
+#include <string>
+
+namespace ccf
+{
+  struct EntityId
+  {
+    // The underlying value type should be blit-serialisable so that it can be
+    // written to and read from the ring buffer
+    static constexpr size_t LENGTH = 64; // hex-encoded SHA-256 hash
+
+    using Value =
+      std::string; // < hex-encoded hash of caller's DER-encoded certificate
+    Value id;
+
+    EntityId() = default;
+    EntityId(const Value& id_) : id(id_) {}
+
+    void operator=(const EntityId& other)
+    {
+      id = other.id;
+    }
+
+    void operator=(const Value& id_)
+    {
+      id = id_;
+    }
+
+    bool operator==(const EntityId& other) const
+    {
+      return id == other.id;
+    }
+
+    bool operator!=(const EntityId& other) const
+    {
+      return !(*this == other);
+    }
+
+    bool operator<(const EntityId& other) const
+    {
+      return id < other.id;
+    }
+
+    operator std::string() const
+    {
+      return id;
+    }
+
+    auto& value()
+    {
+      return id;
+    }
+
+    auto& value() const
+    {
+      return id;
+    }
+
+    auto data() const
+    {
+      return id.data();
+    }
```

Comment:
```suggestion
    Value& value()
    {
      return id;
    }

    const Value& value() const
    {
      return id;
    }

    char const* data() const
    {
      return id.data();
    }
```

It wasn't obvious to me that `auto& foo() const` was returning a `const Foo&` (rather than the dangerously broken `Foo&`-from-a-const-method), so I think we should spell this ourselves in the return types. And if we intend to use these for all accesses, `id` should be `protected`.